From e15b112e3bb5952f991161a32568d925fa98bbde Mon Sep 17 00:00:00 2001
From: Leigh Scott <leigh123linux@gmail.com>
Date: Tue, 16 Dec 2025 14:34:36 +0000
Subject: [PATCH] Menu: Removes the icon validation logic that only checked
 Gtk.IconTheme

Removed unused icon handling logic from the applet constructor.
---
 .../cinnamon/applets/menu@cinnamon.org/applet.js    | 13 -------------
 1 file changed, 13 deletions(-)

diff --git a/files/usr/share/cinnamon/applets/menu@cinnamon.org/applet.js b/files/usr/share/cinnamon/applets/menu@cinnamon.org/applet.js
index d4564e240d..06ba33a21b 100644
--- files/usr/share/cinnamon/applets/menu@cinnamon.org/applet.js
+++ files/usr/share/cinnamon/applets/menu@cinnamon.org/applet.js
@@ -582,19 +582,6 @@ class ApplicationButton extends GenericApplicationButton {
         super(applet, app, 'app', true, 'appmenu-application-button');
         this.category = [];
         this.icon = this.app.create_icon_texture(applet.applicationIconSize);
-        let gicon = this.icon.get_gicon();
-        if (gicon?.get_names) {
-            let iconNames = gicon.get_names();
-            let iconTheme = Gtk.IconTheme.get_default();
-            let hasAnyIcon = gicon.get_names().some(name => iconTheme.has_icon(name));
-            if (!hasAnyIcon) {
-                this.icon = new St.Icon({
-                    icon_name: 'application-x-executable',
-                    icon_size: applet.applicationIconSize,
-                    icon_type: St.IconType.FULLCOLOR
-                });
-            }
-        }
         this.addActor(this.icon);
 
         this.addLabel(this.name, 'appmenu-application-button-label');